drivers/syslog/ramlog.c

```diff
@@ -337,6 +337,13 @@ static ssize_t ramlog_addbuf(FAR struct ramlog_dev_s *priv,
 
   if (len > 0)
     {
+      /* Lock the scheduler do NOT switch out */
+
+      if (!up_interrupt_context())
+        {
+          sched_lock();
+        }
```

Comment:
Merge these 2 commits. The other one will not work without the other. The CI will also break If submitted separately